CacheMultimap.adoc

```diff
@@ -0,0 +1,79 @@
+= Native cache Mutimap
+
+The goal is to provide Infinispan Native CacheMultimap support. The proposal here is to start with the smallest interface implementation, in order to provide a proper API to the first multimap client, vert-x project.
+However, this API will evolve, and this first implementation must work on Embedded and Server mode.
+This first Multimap won't support duplicate values on the same key. 
+
+[source, java]
+.CacheMultimap.java
+----
+public interface CacheMultimap<K, V> {
+   boolean put(K key, V value); // <1> <2>
+
+   Collection<V> get(K key); // <3>
+
+   boolean remove(K key); //<4>
```

Comment:
The choice doesn't have to be to suffer with a Map biased interface or be hostile towards the Collections Framework like JCache is. You could provide an idiomatic API with no needless return types and an asMap() view as an escape hatch. That has worked well in Guava/Caffeine and avoids a single large, complex Interface that isn't idiomatic with the rest of Java.